mirrorlist/mirrorlist_client.wsgi

```diff
@@ -123,12 +123,14 @@ def request_setup(environ, request):
     return d
 
 
-def keep_only_http_results(input):
-    output = []
+def get_first_http_url(input):
+    """ Only used for the redirect case. In the case
+    a redirect has been requested only the first URL
+    is returned starting with 'http'."""
     for hostid, url in input:
-        if url.startswith(u'http'):
-            output.append((hostid, url))
-    return output
+        if url[0].startswith(u'http'):
+            return url[0]
```

Comment:
I was also thinking about it and it is probably the right thing to do. I was not doing it as there usually always is a mirror with at least one HTTP URL. But now it sounds like a bad idea to rely on that assumption. I will update the PR.